Source/Core/DolphinQt/Config/ARCodeWidget.cpp

```diff
@@ -83,6 +92,40 @@ void ARCodeWidget::OnItemChanged(QListWidgetItem* item)
   if (!m_restart_required)
     ActionReplay::ApplyCodes(m_ar_codes);
 
+  UpdateList();
+  SaveCodes();
+}
+
+void ARCodeWidget::OnContextMenuRequested()
+{
+  QMenu menu;
+
+  menu.addAction(tr("Sort Alphabetically"), this, &ARCodeWidget::SortAlphabetically);
+
+  menu.exec(QCursor::pos());
+}
+
+void ARCodeWidget::SortAlphabetically()
+{
+  m_code_list->sortItems();
+  OnListReordered();
+}
+
+void ARCodeWidget::OnListReordered()
+{
+  // Reorder codes based on the indices of table item
+  std::vector<ActionReplay::ARCode> codes;
+  codes.reserve(m_ar_codes.size());
+
+  for (int i = 0; i < m_code_list->count(); i++)
+  {
+    const int index = m_code_list->item(i)->data(Qt::UserRole).toInt();
+
+    codes.push_back(m_ar_codes[index]);
```

Comment:
```suggestion
    codes.push_back(m_ar_codes[index]);
    codes.push_back(std::move(m_ar_codes[index]));
```
To avoid copying a string and a vector each time?